source/opt/basic_block.cpp

```diff
@@ -109,6 +109,28 @@ void BasicBlock::ForEachSuccessorLabel(
   }
 }
 
+void BasicBlock::ForEachSuccessorLabel(
+    const std::function<void(uint32_t*)>& f) {
+  auto br = &insts_.back();
+  switch (br->opcode()) {
+    case SpvOpBranch: {
+      uint32_t tmp_id = br->GetOperand(0).words[0];
```

Comment:
@s-perron @alan-baker That would yield a `const uint32_t*` type (because GetOperand return a `const Operand&`  ) so that's incompatible with the type of `f`. Alternatively, a call to `ForEachInId` call can be made instead.

I added this member function to be able to redirect branches in an explicit manner. This is equivalent to `bb->tail()->ForEachInId` for most of the cases, the difference is some ids won't be labels.